### PR TITLE
Measures are continuous from below: general case

### PR DESCRIPTION
**Pull request by glacode**
Now that ` ~~>* ` is available, we can state continuity from below for both the bounded and the unbounded case (with a single notion of convergence).

( Changed comments for Lemmas, as suggested in https://github.com/metamath/set.mm/pull/2479 )

### CODE REVIEW

**Comment by glacode** [comment]
I forgot to write that's mathbox only

File: set.mm [hunk 1]
```diff
@@ -659696,6 +659696,123 @@ not even needed (it can be any class).  (Contributed by Glauco
...
+      OVGVLPVCVICUJZUKULIUMAVCDESUNLUALZMZNZVCSUOLZCOZVGTUPZQAVIVQMZNZVISUOLZCO
+      ZVLTUPZQBKWDWGBAWCBFWCBUBUCBWFVLTBWECGBWERUDBTRVNUQUFVOVSWDWBWGVOVRWCABKV
+      QUGUIVOWAWFVGVLTVOVTWECVCVISUOVAURVPUSULJUMVB $.
+  $}
+
+  ${
+    $d A x $.
+    $( An extended real is plus infinity iff it's larger than all real numbers.
+       (Contributed by Glauco Siliprandi, 13-Feb-2022.) $)
+    xrpnf $p |- ( A e. RR* -> ( A = +oo <-> A. x e. RR x <_ A ) ) $=
```
Comment (benjub) [inline set.mm:659802]: this should probably go to Main if it is not there yet

File: set.mm [hunk 2]
```diff
@@ -698706,6 +698823,125 @@ sequence of increasing measurable sets (with uniformly bounded measure)
...
+    meaiunincf.n $e |- ( ph -> N e. ZZ ) $.
+    meaiunincf.z $e |- Z = ( ZZ>= ` N ) $.
+    meaiunincf.e $e |- ( ph -> E : Z --> dom M ) $.
+    meaiunincf.i $e |- ( ( ph /\ n e. Z ) -> ( E ` n ) C_ ( E ` ( n + 1 ) ) )
+                   $.
+    meaiunincf.x $e |- ( ph -> E. x e. RR A. n e. Z ( M ` ( E ` n ) ) <_ x )
+                  $.
+    meaiunincf.s $e |- S = ( n e. Z |-> ( M ` ( E ` n ) ) ) $.
+    $( Measures are continuous from below (bounded case): if ` E ` is a
+       sequence of non-decreasing measurable sets (with bounded measure) then
```
Comment (benjub) [inline set.mm:698840]: ... is a nondecreasing sequence of measurable sets with bounded measure, then...
(next statement: replace "bounded" by "finite")

**Review by benjub — APPROVED** [review]
Approved with suggestions for some comments.
Is it really necessary to prove each time a version with DV conditions and a version with F/ hypotheses ?  The latter probably suffices, for these more advanced results.

**Comment by glacode** [comment]
> Is it really necessary to prove each time a version with DV conditions
> and a version with F/ hypotheses ? The latter probably suffices, for these more advanced results.

The *v version would become a lemma of the second theorem, thus there would not be a real advantage.

And making *v its own theorem gives the advantage to save an hyp, when dj vars are not an issue.

**Review by avekens — APPROVED** [review]

**Comment by benjub** [comment]
Can you see, @glacode, what happens if you "inline" the *v version in the general version ? If the resulting proof is not overly complex, then it might be enough to keep that proof.  In Metamatn, type "MM> help expand" to see how this works.  First, "expand xxxv" and then, "minimize */except xxxv".  Maybe @avekens has something to add, from his experience.

And what about my other suggestions ? Do you prefer that we merge now and deal with them later (if at all) ?

**Comment by avekens** [comment]
> Can you see, @glacode, what happens if you "inline" the *v version in the general version ? If the resulting proof is not overly complex, then it might be enough to keep that proof. In Metamatn, type "MM> help expand" to see how this works. First, "expand xxxv" and then, "minimize */except xxxv". Maybe @avekens has something to add, from his experience.

I use the versions with F/ hypotheses only in special cases, and I prefer to use the versions with DV conditions in general. Therefore, I propose to provide the versions with F/ hypotheses only if they are actually necessary  for a proof. There is no problem if they are provided in a mathbox (if not used yet), but can be moved to main if required.

**Comment by glacode** [comment]
I use whatever shortens a proof.
As an example, day in and day out I use rexlimdv (when I can) and rexlimd (when I have to). rexlimdv gives a shorter proof, but rexlimd is more general.

**Review by tirix — APPROVED** [review]

**Review by tirix — APPROVED** [review]

**Comment by tirix** [comment]
I'm like @avekens , I try to avoid `\F` as much as possible, and use it only if there is no other choice.
I would not systematically add them.

For example, @glacode , do you have in mind a use for `~meaiuninc3` where they will be needed?
Neither `~meaiuninc` nor `~meaiuninc2` have such non-free conditions.

Anyway, no problem to merge.

**Comment by benjub** [comment]
Since it's mathbox-only, there is little risk of merge conflicts. @glacode do you want to take into account my suggestions above for the comments (finite/bounded measure), and/or @tirix's, or do you prefer that we merge now ?

**Comment by glacode** [comment]
I've already applied changes in the branch of my local repo. Checking diffs would be hard.

I guess the best option is to merge a9f5493

**Comment by glacode** [comment]
> For example, @glacode , do you have in mind a use for `~meaiuninc3` where they will be needed?

@tirix this would deserve a long post on the forum. I'll try to be supershort (I'm sure you can fill in the details):

1. the ` F/_ n E ` condition will immediately be used to easily derive the following form of the theorem (that's the "useful" one)
```metamath
|- ( ( ph /\ n e. Z ) -> B e. dom M )
```
because the main use of the theorem will be something like this example
```metamath
|- ( ph -> ( M ` RR ) = ( ~~>* ` ( k e. Z |-> ( -u k (,) k ) ) ) )
```

2.  |-   ` F/ n ph ` condition makes it easy to use the theorem when ` ph ` contains a  ` A. n e. Z B e. dom M ` subformula and / or a `` A. n e. Z ( E ` n ) C_ ( E ` ( n + 1 ) )  `` subformula (and it will very often be the case)

Couldn't someone do a cbv* in the "caller" ? Sure, but with a single use of the theorem, you will have already added the "code" that I've used to write it once and for all.

As a further note, the next generation of Proof Assistants will be Language Servers, and then it will be "easy" to add a Diagnostic with a QuickFix that invokes the F/_ version of the theorem to fix a dj var constraint violation.

In my opinion, it is underestimated the importance to have theorems with the (reasonable) smallest number of dj constraints (for the newcomer, this is the most difficult part of metamath to grasp; mathematicians consider "non-free w.r.t. a variable" as a given).

> Neither `~meaiuninc` nor `~meaiuninc2` have such non-free conditions.

As you can see, ~ meaiunincf  was in fact part of this PR.

BR
Glauco

**Pull request merged**